File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.py
from collections import deque
import copy
def solution(queue1, queue2):
    queue1 = deque(queue1)
    queue2 = deque(queue2)
    sum_q1 = sum(queue1)
    sum_q2 = sum(queue2)
    answer = 0
    target = (sum(queue1) + sum(queue2)) // 2
    br = 0
    while True:
        if br == 600000:
            answer = -1
            break
        if sum_q1 == sum_q2:
            break
        if sum_q1 > sum_q2:
            temp = queue1.popleft()
            queue2.append(temp)
            sum_q1 -= temp
            sum_q2 += temp
            answer += 1
        else:
            temp = queue2.popleft()
            queue1.append(temp)
            sum_q2 -= temp
            sum_q1 += temp
            answer += 1
        br += 1
    return answer

File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.java
import java.util.*;

class Solution {
    public int solution(int[] queue1, int[] queue2) {
        Queue<Long> q1 = new ArrayDeque<>();
        Queue<Long> q2 = new ArrayDeque<>();
        long qSum1 = 0;
        long qSum2 = 0;
        
        for (int i : queue1) {
            q1.offer((long) i);
            qSum1 += (long) i;
        }
        
        for (int i : queue2) {
            q2.offer((long) i);
            qSum2 += (long) i;
        }
        
        int answer = 0;
        long target = (qSum1 + qSum2) / 2;
        int limit = queue1.length * 3;
        
        while (answer < limit) {
            if (qSum1 == target) break;
            
            if (qSum1 > qSum2) {
                long temp = q1.poll();
                q2.offer(temp);
                qSum1 -= temp;
                qSum2 += temp;
            } else {
                long temp = q2.poll();
                q1.offer(temp);
                qSum2 -= temp;
                qSum1 += temp;
            }
            answer++;
        }
        
        if (answer == limit) answer = -1;
        
        return answer;
    }
}